apps/nextjs/src/hooks/useLessonPlanScrollManagement.ts

```diff
@@ -0,0 +1,155 @@
+import { useCallback, useEffect, useRef, useState } from "react";
```

Comment:
A new hook that handles scrolling behaviour during streaming.

As new sections are completed it scrolls them into view using a queue.